utils/modular_model_converter.py

```diff
@@ -508,6 +522,36 @@ def replace_call_to_super(class_finder: ClassFinder, updated_node: cst.ClassDef,
 }
 
 
+def get_new_part(class_name, base_class):
+    """
+    When MyClassNameAttention inherits from MistralAttention, we need
+    to process the name to properly find dependencies.
+
+    Here we take what is the same (Attention) and what is different
+    when finding the dependencies.
+    """
+    # Find the common suffix between the two class names
+    common_suffix_len = 0
+
+    # Compare the strings from the end to find the length of the common suffix
+    for i in range(1, min(len(class_name), len(base_class)) + 1):
+        if class_name[-i] == base_class[-i]:
+            common_suffix_len += 1
+        else:
+            break
+
+    # Remove the common suffix from the class_name
+    if common_suffix_len > 0:
+        new_part = class_name[:-common_suffix_len]
+    else:
+        new_part = class_name
+
+    # Convert the remaining new part to snake_case
+    snake_case = re.sub(r"(?<!^)(?=[A-Z])", "_", new_part).lower()
+
+    return snake_case
```

Comment:
That's very nice! It means new models will always have snake casing right?

Reply:
Should be yes!